packages/vite/src/node/optimizer/optimizer.ts

```diff
@@ -743,84 +742,56 @@ function setupOnCrawlEnd(onCrawlEnd: () => void): CrawlEndFinder {
   // If all the inputs are dependencies, we aren't going to get any
   // delayDepsOptimizerUntil(id) calls. We need to guard against this
   // by forcing a rerun if no deps have been registered
+  let firstRunEnsured = false
   function ensureFirstRun() {
     if (!firstRunEnsured && seenIds.size === 0) {
       setTimeout(() => {
         if (seenIds.size === 0) {
           callOnCrawlEnd()
         }
-      }, runOptimizerIfIdleAfterMs)
+      }, 200)
     }
     firstRunEnsured = true
   }
 
   function registerWorkersSource(id: string): void {
     workersSources.add(id)
+
     // Avoid waiting for this id, as it may be blocked by the rollup
     // bundling process of the worker that also depends on the optimizer
-    registeredIds = registeredIds.filter((registered) => registered.id !== id)
+    registeredIds.delete(id)
 
-    const resolve = waitingOn.get(id)
-    // Forced resolve to avoid waiting for the bundling of the worker to finish
-    resolve?.()
+    checkIfCrawlEndAfterTimeout()
   }
 
   function delayDepsOptimizerUntil(id: string, done: () => Promise<any>): void {
     if (!seenIds.has(id)) {
       seenIds.add(id)
-      registeredIds.push({ id, done })
-      callOnCrawlEndWhenIdle()
+      if (!workersSources.has(id)) {
+        registeredIds.add(id)
+        done()
+          .catch(() => {})
```

Comment:
Because of this `catch`, which has been around for a long time in different forms, we may be swallowing some load errors during dev. If you remove it, you'll see Vitest complaining about unhandled errors. We need to review this in another PR (this PR doesn't change things but it makes more clear that we are doing it).